### PR TITLE
treewide: unzip buildInputs to nativeBuildInputs (2)

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Helps with cross-compilation. See #112302.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]
not sure about the modules.

**Comment by Ericson2314** [comment]
Based on the comments, I think the modules are fine. Those are a small rebuild if not, too.

File: pkgs/applications/video/kodi/default.nix [hunk 1]
```diff
@@ -177,7 +177,7 @@ in stdenv.mkDerivation {
       libogg libvorbis flac libxslt systemd
       lzo libcdio libmodplug libass libbluray
       sqlite libmysqlclient avahi lame
-      curl bzip2 zip unzip glxinfo
+      curl bzip2 zip glxinfo
```
Comment (jtojnar) [inline pkgs/applications/video/kodi/default.nix:180]: Looks like this has not been removed from the argument list.

Comment (Ericson2314) [reply pkgs/applications/video/kodi/default.nix:180]: on `staging` it is there and unused. On `staging-next it is also in `buildInputs`.

**Pull request merged**